android/examples/unrooted_android.c

```diff
@@ -0,0 +1,300 @@
+    LOGD("    bmAttributes:            %02xh\n", config->bmAttributes);
+    LOGD("    MaxPower:                %u\n", config->MaxPower);
+
+    for (i = 0; i < config->bNumInterfaces; i++)
+        print_interface(&config->interface[i]);
+}
+
+static void print_device(libusb_device *dev, libusb_device_handle *handle)
+{
+    struct libusb_device_descriptor desc;
+    unsigned char string[256];
+    const char *speed;
+    int ret;
+    uint8_t i;
+
+    switch (libusb_get_device_speed(dev)) {
+        case LIBUSB_SPEED_LOW:		speed = "1.5M"; break;
+        case LIBUSB_SPEED_FULL:		speed = "12M"; break;
+        case LIBUSB_SPEED_HIGH:		speed = "480M"; break;
+        case LIBUSB_SPEED_SUPER:	speed = "5G"; break;
+        case LIBUSB_SPEED_SUPER_PLUS:	speed = "10G"; break;
+        default:			speed = "Unknown";
+    }
+
+    ret = libusb_get_device_descriptor(dev, &desc);
+    if (ret < 0) {
+        LOGD("failed to get device descriptor");
+        return;
+    }
+
+    LOGD("Dev (bus %u, device %u): %04X - %04X speed: %s\n",
+           libusb_get_bus_number(dev), libusb_get_device_address(dev),
+           desc.idVendor, desc.idProduct, speed);
+
+    if (!handle)
+        libusb_open(dev, &handle);
+
+    if (handle) {
+        if (desc.iManufacturer) {
+            ret = libusb_get_string_descriptor_ascii(handle, desc.iManufacturer, string, sizeof(string));
+            if (ret > 0)
+                LOGD("  Manufacturer:              %s\n", (char *)string);
+        }
+
+        if (desc.iProduct) {
+            ret = libusb_get_string_descriptor_ascii(handle, desc.iProduct, string, sizeof(string));
+            if (ret > 0)
+                LOGD("  Product:                   %s\n", (char *)string);
+        }
+
+        if (desc.iSerialNumber && verbose) {
+            ret = libusb_get_string_descriptor_ascii(handle, desc.iSerialNumber, string, sizeof(string));
+            if (ret > 0)
+                LOGD("  Serial Number:             %s\n", (char *)string);
+        }
+    }
+
+    if (verbose) {
+        for (i = 0; i < desc.bNumConfigurations; i++) {
+            struct libusb_config_descriptor *config;
+
+            ret = libusb_get_config_descriptor(dev, i, &config);
+            if (LIBUSB_SUCCESS != ret) {
+                LOGD("  Couldn't retrieve descriptors\n");
+                continue;
+            }
+
+            print_configuration(config);
+
+            libusb_free_config_descriptor(config);
+        }
+
+        if (handle && desc.bcdUSB >= 0x0201)
+            print_bos(handle);
+    }
+
+    if (handle)
+        libusb_close(handle);
+}
+
+
+// fileDescriptor = is the native File Descriptor obtained in Java and transfered to native over JNA for Example.
+int unrooted_usb_description(int fileDescriptor)
+{
+    libusb_context *ctx;
+    libusb_device_handle *devh = NULL;
+    int r = 0;
+    verbose = 1;
+    r = libusb_set_option(&ctx, LIBUSB_OPTION_WEAK_AUTHORITY, NULL);
```

Comment:
The signature of `libusb_set_option` in libusb.h is:

`int LIBUSB_CALL libusb_set_option(libusb_context *ctx, enum libusb_option option, ...);`

Shouldn't we instead send ctx which is a pointer and not its address?
This code does not compile :o

Reply:
right, we have to use the Pointer and not it's address.